### DISCUSSION TITLE
Please document how to use capsize with lists

### DISCUSSION THREAD

**Discussion by dabrahams** [seek-oss/capsize#68] [Q&A] [answered]
[this playroom](https://seek-oss.github.io/braid-design-system/playroom/#?code=N4Igxg9gJgpiBcIA8UCWA3ABAZwC4E8AbGAXmGEwAcBDKNAOwHN5MBGABnYBpMBbagB4B1VFFwALFgBZOPfgCdGqeiwA6IdpmoBXXBHWYAvoYB8q%2BpkxJKOAsTIVIhCPLUg0vRvOr4DxsxaWmAAiMLwQmBAAZpgAwtSU2KgAXjCY2klMmPIw1ISYMMS8MPS42JjKSbCRMYkw2lAQ2PDmlkgA9JQBrVZo6AFBvRi2RKTkmABG1GAA1l4Q2vRQboSojOJleeH06jziMGsbLABMmv49gwAKOeioC%2BWFYSW4F1btfd2BbdgwYLh3Fjwowck2mc3kCyWblw2nkAEdtBBUD8-KZXm0%2BuigkD7OQsYMohBSgAxai8VCEfBuABKEAmED0u3xQUJpQAyikYG5WKdKAImV9BkFVvQYAAJA7rXDcgBs7D5AqFQWMWIGSqG6GZllEJHUExghJylx%2BDX0IC1I1xFDQ2EohB80OoE2Iuz41EUygAQgy9Lw3ABadgAOmOAE4AOwAVjCqOZ7TVSqQ2nyOLGFAUShUmHU8v5ICMaMFidWCfVbRsqZBGeUblzroWuBFXOzIFYfJwEFWUFRJgA8roYPJME3sB0usyUBgLTr1NQorhB8b6o11BbK%2BMbXaHS3cE6XSA5O7MwAVCCUAPBgDMxykoZj%2BZVRaC8fMFoxU6fZZnIH1hpgS9NY5V0-dV12tZEtypHc9xgV1q3ob1cF9C8g3YcMpEvKN7wLN9MBfEDBiTQhcMsMCSJZIlcA5VI3EvOsD3IywmwlQ5pRbGQFQYgj1Xg2tOK4cjHzLQZS2EqxVktNM3Q9LMc34yJdCbbl22wTtRB7RjMAAORgPAYCgYciUYQysibNhMAkahcEwAB3bxEnIjoS0cksdL0gznFMozh2UNJjgs8QrNs%2BzR3aZzuLadpk1E9VJ01CKgm-OcF3kADGiA80EtIuwpM3e0oPUXdnVgg9pJPM8UPYAAOHljkjWMEvwsSnNQGLIuirEOg%2BV4Oh%2BP4AU%2BQi%2BkkkEplmeZFmWFtVilbAtiJV19lYk4zlMHSBGsx5ilKLqMDVXb%2BnMA6AgPEAJCebAEAAbXcCAwEugBdLgQBs0QJEu%2BArvDGUqoewwgA) appears to show that it's trickier than it looks, because the `::after` element on a ul takes effect in the wrong place to make the adjustment.

**Comment by michaeltaranto** [comment]
Thanks for the suggestion and will take that on when the documentation gets uplifted next.

In terms of your implementation, Capsize should only be applied to the leaf text nodes. There was a bit of a mix in the link you shared.

Once applied correctly, the outstanding white space was being introduced by the default `marker` element. Capsize cannot style this element due to its [CSS property limitations](https://developer.mozilla.org/en-US/docs/Web/CSS/::marker#allowable_properties). However, I would recommend styling the `marker` element to use a `line-height` that is closer to the cap height.

[Updated Playroom](https://seek-oss.github.io/braid-design-system/playroom/#?code=N4Igxg9gJgpiBcIA8UCWA3ABAZwC4E8AbGAXmGEwAcBDKNAOwHN5MBGABnYBpMBbagB4B1VFFwALFgBZOPfgCdGqeiwA6IdpmoBXXBHWYAvoYB8q%2BpkxJKOAsTIVIhCPLUg0vRvOr4DxsxaWmAAiMLwQmBAAZpgAwtSU2KgAXjBalJSEqDBQmHpamFl48OaWSAD0lAGlVmjoAUG1GLZEpOSYAEbUYADWXhDa9FBuWYziuNjUhOH06jziMKhjuCwATJr%2BNY0ACvIw6KgD2JgwxLww9LhbVuV11YFl2DBguIezD0F4rQ7XjV29-UGw0w6lw2nkAEdtBBUE85r8glEIJcAGLUXioQj4NwAJQgHQgenhH0aSMuAGUUjA3Kx1pQBMTGo0svQYAAJRbLGkANnY9MZTOM1waTOA5QAVJgcTAnrhMLAojpCHKvsRjuLyoYEUhtIQWvZ2golCpMBtTAjLBbMGLJdKoNowGkFktxpEYhI0kU5QoejB5IVlGkCQIAHRxBJJVKYMDUej0Ql5eSoXh5cSwrhWxrYCJ7SPKRh5CKxk4piTUOWw6POJ7%2B-Ie6MJTDO5aFqh7dAXK4kpmWZS4eTQB2vZFuzAAdzTuDS2BojpDVo1Wu7jzsMBMwAABll4PAfX7rSyYABaZvjeAhgDsYUMG8MFVVa6tVqQWRFPcty6CNqlMpgcoVSoqqu6qapmVg2A%2BDh8NQijKCw3BULQDDMKaRjmp%2BjTfgAovQ2Dgmk8SJFSmCVgkmTZLkdYLIUMDUO6MACHK8awCBS7vkyKAYGBjSiCQ6gdDASJ7AA8rofrbE82hQPoIDcZ8q5QWgM6ED4bi4NQHTEHM0GwfQABChJ6LwbhHuwIarAAnBeACsYR%2BGx7FBOUb6OaJU7%2Boe2ByZx6Byb2UB8SAdHuW54mSdJ6h%2BfqbQUEpmSqSCIDqZpMDaUaygACoQJQJlmQAzKsUgWXZIBoXJzlgRUVTmJVupyZB5BRWSuCUqkbh5XyDIgBmGFMoeHIuisiUyPy3VRelJrcHJQq9S57Evqgc2OT5UX%2BYFAlCTAACScZhTAUkQKwkW9T2DWxbC8XYolyVad1OnGgZuBGbl5lWbZvD2VFFUnUEAByMpTrkzhMIUyIFoebCpuW47eIkUUrT9PEBeowV%2BjtrLyBJ%2B3SUdsmI-J3ztHFKlXaCGm3XIMHGllOWJaZIYFUVJVlfj32OUEFSvt5XP4wj7M9rx-GCS4227Zj4UQKsx380yZ3yhdJNqeTqV3RNj3PXTZmWTZzMzTLlhs-r-14DkoMg8D4OBpgqxQ3KY6w15vN1KtJHI0FUTueje0HVLeP6wTBrncpCVkylaVU5l2UvYzxUfaVev64b7Oc4tlXlLqS2NM%2B5Q8z2FQZ9c97PMO9D3I0PnRVB-x9AOQIjINkzTMi2mnkN6xoSY-2MScZydhUdw1P3GABEP9TmN1SULOc2AIAA2u4EBgDPAC6XAgGOogSDP8Czxe3IABzL4YQA)

**Comment by dabrahams** [comment, marked as answer]
> Capsize should only be applied to the leaf text nodes 

That's what I figured, but it would be good to make that very clear in the documentation.  There's not really a good way to select them with CSS selectors, is there?

In my actual application I need to control list bullet formatting using a ::before pseudo-element, which makes applying the standard basekick to the leaf `<li>` elements impractical, which is why it ended up on `<ul>`.  But there are places where I had to make a capsize adjustment for other CSS-injected text, and there I used a solid-trasparent border to do it.  I suppose I could do the same for `<li>` elements…

Looking at your updated playroom, it seems you added a `<p>` around the top-level list item so you could style it individually.  That's a great solution, if you can do it.  Since my lists are being generated with Markdown, I don't have that option without adding javascript injection.  I'd rather not resort to that, but HTML lists are rather lacking in intrinsic structure that allows control over style, so maybe it's the best option.  Any advice appreciated here.

**Reply by michaeltaranto** [reply]
That `p` tag was actually in your original playroom which is why I left it. As you were describing I would suggest leaving the `li` elements separate from capsize so you can style them however you want (with pseudos or built ins etc). Then you apply Capsize to the text node within the list item. That node could be a `a` or a `span` or whatever you choose, but so long as the list and list item are reset in terms of margins and paddings it should work fine.

For example:
```html
<style type="text/css">
  ul, li { margin: 0; padding: 0; }
  .listItem { /* ... */  }
</style>

<ul>
  <li class="listItem">
    <a class="capsizedText">...</a>
  </li>
  <li class="listItem">
    <span class="capsizedText">...</span>
  </li>
</ul>
```

**Comment by dabrahams** [comment]
Huh; I never considered using an `<a>` without an `href` before.  At least that's somewhat minimal.

**Reply by michaeltaranto** [reply]
That wasnt the point of the example code.

**Reply by dabrahams** [reply]
Thanks, I didn't miss the point; I was just remarking. 

This suggests I may need to add HTML structure purely for styling purposes. I suppose there's no alternative until CSS is fixed to deal with this issue… I do hope whatever they have planned accounts for these cases.
